interface/patient_file/summary/demographics.php

```diff
@@ -1223,12 +1224,29 @@ function setMyPatient() {
                     endif;
 
                     if ($GLOBALS['portal_onsite_two_enable']):
+                        /** @var EventDispatcher */
+                        $d = $GLOBALS['kernel']->getEventDispatcher();
+                        $e = new CardRenderEvent('portal');
```

Comment:
Are you thinking of firing a CardRenderEvent for each one of the cards?  Portal here being just one example?  Just trying to make sure I'm following the flow here.

Reply:
Exactly, each card would have a new CardRenderEvent with the card ID being passed to it. CardRenderEvent is generic to all cards. The portal use is just a proof for the discussion